axelor-bank-payment/src/main/java/com/axelor/apps/bankpayment/service/bankreconciliation/BankReconciliationServiceImpl.java

```diff
@@ -1052,12 +1065,29 @@ protected BigDecimal getAmountMarginLow(BankReconciliation bankReconciliation) {
   }
 
   protected Context getScriptContext(
-      BankStatementLine bankStatementLine, BankReconciliationLine bankReconciliationLine) {
+      BankReconciliation bankReconciliation,
+      BankStatementLine bankStatementLine,
+      BankReconciliationLine bankReconciliationLine,
+      MoveLine moveLine)
+      throws AxelorException {
     Context scriptContext =
         new Context(Mapper.toMap(bankStatementLine), BankStatementLineAFB120.class);
 
-    scriptContext.put("debit", bankReconciliationLine.getDebit());
-    scriptContext.put("credit", bankReconciliationLine.getCredit());
+    BigDecimal debit = bankReconciliationLine.getDebit();
+    BigDecimal credit = bankReconciliationLine.getCredit();
+
+    BigDecimal currencyAmount = debit.compareTo(BigDecimal.ZERO) == 0 ? credit : debit;
+    currencyAmount =
+        currencyAmount.divide(
+            currencyService.getCurrencyConversionRate(
+                bankReconciliation.getCurrency(),
+                moveLine.getMove().getCurrency(),
+                dateService.date()),
+            RoundingMode.HALF_UP);
```

Comment:
we should call getAmountCurrencyConvertedAtDate directly instead, the scale is missing here